# An International Cuisine Restaurant in a Toronto city

## 1. Introduction¶

This report can help those who are planning to start a new business such as an international cuisine restaurant in Toronto city and need to find the best location for their shop.

## 1.1 Background

 people's preferences and needs change day by day. For this reason, new shopping malls, shops, restaurants etc. are set up every day all over the world.

The population of Toronto is increasing. New restaurants are needed in this city. There are also different restaurants, which include different cuisine in Toronto.

Especially the central districts of the Toronto are becoming more and more become a shining star

The number of population increases due to newborns and migration. Therefore demands are increasing in every areas.Such as demands of restaurants

Since Toronto is a very important place, it attracts attention from very famous brands. Many famous brands want to be in Toronto. The entrepreneurs want to take advantage by having restaurants in the central of Toronto.



## 1.2 Business Problem

Acceptances:

1. Restaurant customers are located around the available restaurants. And it is appropriate to specify these clients in the make_blobs method with        cluster_std = 0.003.

2. Since it covers the world cuisines, it aspires to the customers of every available restaurant.

3. Since the company wanted to open a place in the center, the information of the restaurants close to the center was used.

4. It is accepted that every restaurant has registered to the site where the data is received.

Since the restaurant is a world brand, it covers the world food and is a competitive company, it will increase the probability of success, rather than escaping from the center. Therefore, I need to find where the customer cluster is most concentrated.

The problem is :If the client want to open an international luxury cuisine restaurant in Toronto city, which point is the best option to open the restaurant?

## 1.3 Interest

Selecting the right location for a business is one of the first and very important decisions in running a business.

This report can help those who are planning to start a new business such as an international luxury cuisine restaurant in Toronto city and need to find the best location for their shop.

A world brand restaurant chain, which can better understand people's new wishes, is innovative, contains many world cuisines and can attract customers from existing restaurants, will want to make a place in Toronto. And they will want to know where the most suitable place is for this place. The customer will look for a location with the highest potential.

The company concerned is a company that wants to appeal to a large group of customers in the center, not to a small group of customers

## 2. Data Acquisition And Cleaning

I will extract the name of the neighborhoods of Toronto from the following link in Wikipedia: 'https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M'

I will help to eliminate bad choice areas for sellecting the most appropriate restaurant for client.

After that, I will use Foursquare API to get the information about the restaurants.The informations get extracted from the following link: 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(

In accordance with the acceptance_1 using the restaurant coordinates, I will find the places where the customers are most intense. I will use cluster_std = 0.004 in the make_blobs method to find them.

I will find the most suitable region with the K-means-clustering method by using the places where the customers are the most crowded. I will find k in the K-means-clustering method that I will use here with estimate_bandwidth.

I will apply the K-Means Clustering method again in this region and find the center of this place. so I will find the best location of the best region.

In [ ]:
import requests
import urllib.request
import time
#conda install -c anaconda beautifulsoup4
from bs4 import BeautifulSoup
import numpy as np
import pandas as pd
from urllib.request import urlopen
import numpy as np # library to handle data in a vectorized manner

import pandas as pd # library for data analsysis
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

import json # library to handle JSON files

#!conda install -c conda-forge geopy --yes # uncomment this line if you haven't completed the Foursquare API lab
from geopy.geocoders import Nominatim # convert an address into latitude and longitude values

import requests # library to handle requests
from pandas.io.json import json_normalize # tranform JSON file into a pandas dataframe

# Matplotlib and associated plotting modules
import matplotlib.cm as cm
import matplotlib.colors as colors

# import k-means from clustering stage
from sklearn.cluster import KMeans

#!conda install -c conda-forge folium=0.5.0 --yes # uncomment this line if you haven't completed the Foursquare API lab
import folium # map rendering library

print('Libraries imported.')

Libraries imported.

In [2]:
import random 
import numpy as np 
import matplotlib.pyplot as plt 
from sklearn.cluster import KMeans 
from sklearn.datasets.samples_generator import make_blobs 
%matplotlib inline

In [3]:
url = 'https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M'
html = urlopen(url) 
soup = BeautifulSoup(html, 'html.parser')

In [4]:
row = soup.find('tbody') # Extract and return first occurrence of tbody
#Let's fix the data in a few commands below.
list_=row.get_text().split("\n")  # we try to get the writings. we do data separation from spaces.
list_[:20]

['',
 '',
 'M1ANot assigned',
 '',
 '',
 '',
 'M2ANot assigned',
 '',
 '',
 '',
 'M3ANorth York(Parkwoods)',
 '',
 '',
 '',
 'M4ANorth York(Victoria Village)',
 '',
 '',
 '',
 'M5ADowntown Toronto(Regent Park / Harbourfront)',
 '']

In [5]:
list_1=[]

for i in list_:
    if i!="":
        if (i[3:])!='Not assigned': #eliminating 'Not assigned' 
            list_1.append(i[3:])
print(list_1[:20])

['North York(Parkwoods)', 'North York(Victoria Village)', 'Downtown Toronto(Regent Park / Harbourfront)', 'North York(Lawrence Manor / Lawrence Heights)', "Queen's Park / Ontario Provincial Government", 'Etobicoke(Islington Avenue)', 'Scarborough(Malvern / Rouge)', 'North York(Don Mills)North', 'East York(Parkview Hill / Woodbine Gardens)', 'Downtown Toronto(Garden District, Ryerson)', 'North York(Glencairn)', 'Etobicoke(West Deane Park / Princess Gardens / Martin Grove / Islington / Cloverdale)', 'Scarborough(Rouge Hill / Port Union / Highland Creek)', 'North York(Don Mills)South(Flemingdon Park)', 'East York(Woodbine Heights)', 'Downtown Toronto(St. James Town)', 'York(Humewood-Cedarvale)', 'Etobicoke(Eringate / Bloordale Gardens / Old Burnhamthorpe / Markland Wood)', 'Scarborough(Guildwood / Morningside / West Hill)', 'East Toronto(The Beaches)']


In [11]:
borough=[]
for i in list_1:
  
    for j in range(len(i)):
        if i[j]=="(":
            borough.append(i[:j])
print(borough[:10])

['North York', 'North York', 'Downtown Toronto', 'North York', 'Etobicoke', 'Scarborough', 'North York', 'East York', 'Downtown Toronto', 'North York']


In [15]:
borough.insert(4,"Queen's Park")
borough[:10]

['North York',
 'North York',
 'Downtown Toronto',
 'North York',
 "Queen's Park",
 "Queen's Park",
 "Queen's Park",
 "Queen's Park",
 'Etobicoke',
 'Scarborough']

In [16]:
df = pd.DataFrame()
df['Borough']=borough
df.head()

Borough
0        North York
1        North York
2  Downtown Toronto
3        North York
4      Queen's Park

In [18]:
df.groupby(['Borough']).size()

Borough
Central Toronto                                                  9
Downtown Toronto                                                17
Downtown TorontoStn A PO Boxes25 The Esplanade                   1
East Toronto                                                     4
East TorontoBusiness reply mail Processing Centre969 Eastern     1
East York                                                        4
East YorkEast Toronto                                            1
Etobicoke                                                       11
EtobicokeNorthwest                                               1
MississaugaCanada Post Gateway Processing Centre                 1
North York                                                      24
North York(Don Mills)South                                       1
North York(Downsview)East                                        1
Queen's Park                                                     4
Scarborough                                           

In [19]:
#Let's correct the misspellings.
d = {'Downtown TorontoStn A PO Boxes25 The Esplanade':'Downtown Toronto',
     'East TorontoBusiness reply mail Processing Centre969 Eastern':'East Toronto',
    'East YorkEast Toronto':'East York',
    'MississaugaCanada Post Gateway Processing Centre':'Mississauga',
    'EtobicokeNorthwest':'Etobicoke','North York(Don Mills)South':'North York',
    'North York(Downsview)East':'North York'}
df = df.replace(d)
df.groupby(['Borough']).size()

Borough
Central Toronto                 9
Downtown Toronto               18
East Toronto                    5
East York                       5
Etobicoke                      12
Mississauga                     1
North York                     25
North York(Downsview)East       1
Queen's Park                    4
Scarborough                    17
West Toronto                    6
York                            5
dtype: int64

In [20]:
list_=[]
for i in df["Borough"]:
    if i not in list_ and i!="North York(Downsview)East  ":
        list_.append(i)
print(list_)
borough=list_

['North York', 'Downtown Toronto', "Queen's Park", 'Etobicoke', 'Scarborough', 'East York', 'York', 'East Toronto', 'West Toronto', 'Central Toronto', 'Mississauga']


In [21]:
df = pd.DataFrame()
df['Borough']=list_
df

Borough
0         North York
1   Downtown Toronto
2       Queen's Park
3          Etobicoke
4        Scarborough
5          East York
6               York
7       East Toronto
8       West Toronto
9    Central Toronto
10       Mississauga

In [23]:
#['Downtown Toronto', 'East Toronto', 'West Toronto', 'Central Toronto']
borough_coordinate_lat=[]
borough_coordinate_long=[]
for i in borough :
    address = [ (i+',Toronto, CA')]
    geolocator = Nominatim(user_agent="foursquare_agent")
    location = geolocator.geocode(address)
    lat = location.latitude
    lng = location.longitude
    borough_coordinate_lat.append(lat)
    borough_coordinate_long.append(lng)
print(borough_coordinate_lat)
print(borough_coordinate_long)

[43.7543263, 43.6563221, 43.659659, 43.671459150000004, 43.773077, 43.699971000000005, 43.67910515, 43.6247901, 43.653963, 43.653963, 43.6668555]
[-79.44911696639593, -79.3809161, -79.3903399, -79.55249206611668, -79.257774, -79.33251996261595, -79.49118414007154, -79.3934918, -79.387207, -79.387207, -79.5879563]


In [24]:
# lets combine Borough ,Latitude and Longitude informations
toronto_data = pd.DataFrame([borough,borough_coordinate_lat,borough_coordinate_long]).T
toronto_data.columns=['Borough', 'Latitude','Longitude']
toronto_data

Borough Latitude Longitude
0         North York  43.7543  -79.4491
1   Downtown Toronto  43.6563  -79.3809
2       Queen's Park  43.6597  -79.3903
3          Etobicoke  43.6715  -79.5525
4        Scarborough  43.7731  -79.2578
5          East York     43.7  -79.3325
6               York  43.6791  -79.4912
7       East Toronto  43.6248  -79.3935
8       West Toronto   43.654  -79.3872
9    Central Toronto   43.654  -79.3872
10       Mississauga  43.6669   -79.588

In [30]:
address = 'Toronto, CA'

geolocator = Nominatim(user_agent="ny_explorer")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print('The geograpical coordinate of Toronto City are {}, {}.'.format(latitude, longitude))

The geograpical coordinate of Toronto City are 43.653963, -79.387207.


### Since our company will create a place in the center, let's eliminate it in small and remote places.

In [26]:
CLIENT_ID = '...............................' # your Foursquare ID
CLIENT_SECRET = '...........................' # your Foursquare Secret
VERSION = '20180605' # Foursquare API version

print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: ...............................
CLIENT_SECRET:...........................


In [28]:
def getNearbyVenues(names, latitudes, longitudes, radius=30000):
    LIMIT=10
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Neighborhood', 
                  'Neighborhood Latitude', 
                  'Neighborhood Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    print(url,results,venues_list,nearby_venues,sep="\n")
    
    return(nearby_venues)

In [ ]:
toronto_venues = getNearbyVenues(names=toronto_data['Borough'],
                                   latitudes=toronto_data['Latitude'],
                                   longitudes=toronto_data['Longitude']
                                  )
print("done")


North York,
Downtown Toronto,
Queen's Park,
Etobicoke,
Scarborough,
East York,
York,
East Toronto,
West Toronto,
Central Toronto,
Mississauga.
https://api.foursquare.com/v2/venues/explore?&client_id=3DY3IKVXYZDQFTBN3XEPQW2RWHSPCCFM0DRSLPS3YK1Z513O&client_secret=APLJQGA24KCQVXBBRG0ZNJ5EK2TWR21SCTL0VG2WLAXJ5PNY&v=20180605&ll=43.6668555,-79.5879563&radius=30000&limit=10
[{'reasons': {'count': 0, 'items': [{'summary': 'This spot is popular', 'type': 'general', 'reasonName': 'globalInteractionReason'}]}, 'venue': {'id': '4af4709bf964a52079f221e3', 'name': 'Porta Via', 'location': {'address': '5399 Eglinton Avenue West, Unit 104', 'crossStreet': 'btw Renforth & Commerce', 'lat': 43.66344890472386, 'lng': -79.58963815332794, 'labeledLatLngs': [{'label': 'display', 'lat': 43.66344890472386, 'lng': -79.58963815332794}], 'distance': 402, 'cc': 'CA', 'city': 'Toronto', 'state': 'ON', 'country': 'Canada', 'formattedAddress': ['5399 Eglinton Avenue West, Unit 104 (btw Renforth & Commerce)', 'Toronto ON', 'Canada']}, 'categories': [{'id': '4bf58dd8d48988d1c5941735', 'name': 'Sandwich Place', 'pluralName': 'Sandwich Places', 'shortName': 'Sandwiches', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/deli_', 'suffix': '.png'}, 'primary': True}], 'photos': {'count': 0, 'groups': []}}, 'referralId': 'e-0-4af4709bf964a52079f221e3-0'}, {'reasons': {'count': 0, 'items': [{'summary': 'This spot is popular', 'type': 'general', 'reasonName': 'globalInteractionReason'}]}, 'venue': {'id': '4bcf2c40caff95211a9bccf0', 'name': 'Centennial Park', 'location': {'address': '256 Centennial Park Rd.', 'lat': 43.6561537571146, 'lng': -79.58753988924185, 'labeledLatLngs': [{'label': 'display', 'lat': 43.6561537571146, 'lng': -79.58753988924185}], 'distance': 1191, 'postalCode': 'M9C 5N3', 'cc': 'CA', 'city': 'Etobicoke', 'state': 'ON', 'country': 'Canada', 'formattedAddress': ['256 Centennial Park Rd.', 'Etobicoke ON M9C 5N3', 'Canada']}, 'categories': [{'id': '4bf58dd8d48988d163941735', 'name': 'Park', 'pluralName': 'Parks', 'shortName': 'Park', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/parks_outdoors/park_', 'suffix': '.png'}, 'primary': True}], 'photos': {'count': 0, 'groups': []}}, 'referralId': 'e-0-4bcf2c40caff95211a9bccf0-1'}, {'reasons': {'count': 0, 'items': [{'summary': 'This spot is popular', 'type': 'general', 'reasonName': 'globalInteractionReason'}]}, 'venue': {'id': '572de8f6498eb9a0687aa5e4', 'name': 'Indigo', 'location': {'address': '25 The West Mall', 'crossStreet': 'at The Queensway', 'lat': 43.61336504116866, 'lng': -79.55798001820008, 'labeledLatLngs': [{'label': 'display', 'lat': 43.61336504116866, 'lng': -79.55798001820008}], 'distance': 6425, 'postalCode': 'M9C 1B8', 'cc': 'CA', 'city': 'Etobicoke', 'state': 'ON', 'country': 'Canada', 'formattedAddress': ['25 The West Mall (at The Queensway)', 'Etobicoke ON M9C 1B8', 'Canada']}, 'categories': [{'id': '4bf58dd8d48988d114951735', 'name': 'Bookstore', 'pluralName': 'Bookstores', 'shortName': 'Bookstore', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/shops/bookstore_', 'suffix': '.png'}, 'primary': True}], 'photos': {'count': 0, 'groups': []}}, 'referralId': 'e-0-572de8f6498eb9a0687aa5e4-2'}, {'reasons': {'count': 0, 'items': [{'summary': 'This spot is popular', 'type': 'general', 'reasonName': 'globalInteractionReason'}]}, 'venue': {'id': '4b846845f964a520ad3331e3', 'name': 'Kingsway Fish & Chips', 'location': {'address': '3060 Bloor St W', 'crossStreet': 'at Royal York Rd.', 'lat': 43.647074, 'lng': -79.513291, 'labeledLatLngs': [{'label': 'display', 'lat': 43.647074, 'lng': -79.513291}], 'distance': 6403, 'postalCode': 'M8X 1C4', 'cc': 'CA', 'neighborhood': 'The Kingsway', 'city': 'Etobicoke', 'state': 'ON', 'country': 'Canada', 'formattedAddress': ['3060 Bloor St W (at Royal York Rd.)', 'Etobicoke ON M8X 1C4', 'Canada']}, 'categories': [{'id': '4bf58dd8d48988d1ce941735', 'name': 'Seafood Restaurant', 'pluralName': 'Seafood Restaurants', 'shortName': 'Seafood', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/seafood_', 'suffix': '.png'}, 'primary': True}], 'photos': {'count': 0, 'groups': []}}, 'referralId': 'e-0-4b846845f964a520ad3331e3-3'}, {'reasons': {'count': 0, 'items': [{'summary': 'This spot is popular', 'type': 'general', 'reasonName': 'globalInteractionReason'}]}, 'venue': {'id': '56c8dbe9498e7033010ac0e5', 'name': 'JOEY Sherway', 'location': {'address': '25 The West Mall,Unit 500', 'crossStreet': 'The Queensway', 'lat': 43.61309535370009, 'lng': -79.55638480409485, 'labeledLatLngs': [{'label': 'display', 'lat': 43.61309535370009, 'lng': -79.55638480409485}], 'distance': 6502, 'postalCode': 'M9C 1B8', 'cc': 'CA', 'neighborhood': 'Islington - City Centre West', 'city': 'Toronto', 'state': 'ON', 'country': 'Canada', 'formattedAddress': ['25 The West Mall,Unit 500 (The Queensway)', 'Toronto ON M9C 1B8', 'Canada']}, 'categories': [{'id': '4bf58dd8d48988d1c4941735', 'name': 'Restaurant', 'pluralName': 'Restaurants', 'shortName': 'Restaurant', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/default_', 'suffix': '.png'}, 'primary': True}], 'photos': {'count': 0, 'groups': []}}, 'referralId': 'e-0-56c8dbe9498e7033010ac0e5-4'}, {'reasons': {'count': 0, 'items': [{'summary': 'This spot is popular', 'type': 'general', 'reasonName': 'globalInteractionReason'}]}, 'venue': {'id': '4ba3fa87f964a520d17338e3', 'name': 'SanRemo Bakery', 'location': {'address': '374 Royal York Rd', 'crossStreet': 'at Simpson Ave', 'lat': 43.618542136521064, 'lng': -79.49948481145465, 'labeledLatLngs': [{'label': 'display', 'lat': 43.618542136521064, 'lng': -79.49948481145465}], 'distance': 8928, 'postalCode': 'M8Y 2R3', 'cc': 'CA', 'city': 'Toronto', 'state': 'ON', 'country': 'Canada', 'formattedAddress': ['374 Royal York Rd (at Simpson Ave)', 'Toronto ON M8Y 2R3', 'Canada']}, 'categories': [{'id': '4bf58dd8d48988d16a941735', 'name': 'Bakery', 'pluralName': 'Bakeries', 'shortName': 'Bakery', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/bakery_', 'suffix': '.png'}, 'primary': True}], 'photos': {'count': 0, 'groups': []}}, 'referralId': 'e-0-4ba3fa87f964a520d17338e3-5'}, {'reasons': {'count': 0, 'items': [{'summary': 'This spot is popular', 'type': 'general', 'reasonName': 'globalInteractionReason'}]}, 'venue': {'id': '55fd4770498e7a8fbc9082fe', 'name': 'Starbucks', 'location': {'address': '829 The Queensway', 'crossStreet': 'Taymall Avenue', 'lat': 43.624654, 'lng': -79.508217, 'labeledLatLngs': [{'label': 'display', 'lat': 43.624654, 'lng': -79.508217}], 'distance': 7957, 'postalCode': 'M8Z 1N6', 'cc': 'CA', 'city': 'Toronto', 'state': 'ON', 'country': 'Canada', 'formattedAddress': ['829 The Queensway (Taymall Avenue)', 'Toronto ON M8Z 1N6', 'Canada']}, 'categories': [{'id': '4bf58dd8d48988d1e0931735', 'name': 'Coffee Shop', 'pluralName': 'Coffee Shops', 'shortName': 'Coffee Shop', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/coffeeshop_', 'suffix': '.png'}, 'primary': True}], 'photos': {'count': 0, 'groups': []}}, 'referralId': 'e-0-55fd4770498e7a8fbc9082fe-6'}, {'reasons': {'count': 0, 'items': [{'summary': 'This spot is popular', 'type': 'general', 'reasonName': 'globalInteractionReason'}]}, 'venue': {'id': '4ad4c05ef964a5209ff620e3', 'name': 'High Park', 'location': {'address': '1873 Bloor St. W', 'crossStreet': 'at Parkside Dr.', 'lat': 43.6464791690415, 'lng': -79.46342468261719, 'labeledLatLngs': [{'label': 'display', 'lat': 43.6464791690415, 'lng': -79.46342468261719}], 'distance': 10282, 'postalCode': 'M6R 2Z3', 'cc': 'CA', 'city': 'Toronto', 'state': 'ON', 'country': 'Canada', 'formattedAddress': ['1873 Bloor St. W (at Parkside Dr.)', 'Toronto ON M6R 2Z3', 'Canada']}, 'categories': [{'id': '4bf58dd8d48988d163941735', 'name': 'Park', 'pluralName': 'Parks', 'shortName': 'Park', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/parks_outdoors/park_', 'suffix': '.png'}, 'primary': True}], 'photos': {'count': 0, 'groups': []}}, 'referralId': 'e-0-4ad4c05ef964a5209ff620e3-7'}, {'reasons': {'count': 0, 'items': [{'summary': 'This spot is popular', 'type': 'general', 'reasonName': 'globalInteractionReason'}]}, 'venue': {'id': '4af5b68cf964a520c4fb21e3', 'name': 'Humber Bay Park', 'location': {'address': 'Humber Bay Park East', 'crossStreet': 'Lakeshore & Park Lawn', 'lat': 43.6223956417084, 'lng': -79.47838891858714, 'labeledLatLngs': [{'label': 'display', 'lat': 43.6223956417084, 'lng': -79.47838891858714}], 'distance': 10119, 'cc': 'CA', 'city': 'Toronto', 'state': 'ON', 'country': 'Canada', 'formattedAddress': ['Humber Bay Park East (Lakeshore & Park Lawn)', 'Toronto ON', 'Canada']}, 'categories': [{'id': '4bf58dd8d48988d163941735', 'name': 'Park', 'pluralName': 'Parks', 'shortName': 'Park', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/parks_outdoors/park_', 'suffix': '.png'}, 'primary': True}], 'photos': {'count': 0, 'groups': []}}, 'referralId': 'e-0-4af5b68cf964a520c4fb21e3-8'}, {'reasons': {'count': 0, 'items': [{'summary': 'This spot is popular', 'type': 'general', 'reasonName': 'globalInteractionReason'}]}, 'venue': {'id': '540f2951498e7516839a7126', 'name': 'nodo', 'location': {'address': '2885\xa0Dundas\xa0St.\xa0West', 'lat': 43.665303321627505, 'lng': -79.46562068004896, 'labeledLatLngs': [{'label': 'display', 'lat': 43.665303321627505, 'lng': -79.46562068004896}], 'distance': 9852, 'cc': 'CA', 'city': 'Toronto', 'state': 'ON', 'country': 'Canada', 'formattedAddress': ['2885\xa0Dundas\xa0St.\xa0West', 'Toronto ON', 'Canada']}, 'categories': [{'id': '4bf58dd8d48988d110941735', 'name': 'Italian Restaurant', 'pluralName': 'Italian Restaurants', 'shortName': 'Italian', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/italian_', 'suffix': '.png'}, 'primary': True}], 'photos': {'count': 0, 'groups': []}}, 'referralId': 'e-0-540f2951498e7516839a7126-9'}]
[[('North York', 43.7543263, -79.44911696639593, 'Earl Bales Park', 43.75304340101652, -79.43622849867882, 'Park'), ('North York', 43.7543263, -79.44911696639593, 'Nordstrom', 43.7260545, -79.4494431, 'Clothing Store'), ('North York', 43.7543263, -79.44911696639593, 'Maryam Hotel', 43.766961, -79.401199, 'Hotel'), ('North York', 43.7543263, -79.44911696639593, 'Kinka Izakaya', 43.76016102214242, -79.40982686116466, 'Japanese Restaurant'), ('North York', 43.7543263, -79.44911696639593, 'Crate & Barrel', 43.72658365, -79.45266115, 'Furniture / Home Store'), ('North York', 43.7543263, -79.44911696639593, 'Auberge du Pommier', 43.74696232135214, -79.407878758257, 'French Restaurant'), ('North York', 43.7543263, -79.44911696639593, 'Amadeus Fine Cakes', 43.80258190023286, -79.44874257686705, 'Dessert Shop'), ('North York', 43.7543263, -79.44911696639593, 'The Abbot', 43.703687730512435, -79.41348481516249, 'Gastropub'), ('North York', 43.7543263, -79.44911696639593, 'Cedarvale Park', 43.692534923091934, -79.42870527613704, 'Field'), ('North York', 43.7543263, -79.44911696639593, 'Sherwood Park', 43.71655100307589, -79.38777567141624, 'Park')], [('Downtown Toronto', 43.6563221, -79.3809161, 'Downtown Toronto', 43.65323167517444, -79.38529600606677, 'Neighborhood'), ('Downtown Toronto', 43.6563221, -79.3809161, 'UNIQLO ユニクロ', 43.65591027779457, -79.38064099181345, 'Clothing Store'), ('Downtown Toronto', 43.6563221, -79.3809161, 'Hailed Coffee', 43.65883296982352, -79.38368351986598, 'Coffee Shop'), ('Downtown Toronto', 43.6563221, -79.3809161, 'Richmond Station', 43.65156872562108, -79.37926594000328, 'American Restaurant'), ('Downtown Toronto', 43.6563221, -79.3809161, 'Fahrenheit Coffee', 43.65238358726612, -79.37271903848271, 'Coffee Shop'), ('Downtown Toronto', 43.6563221, -79.3809161, 'Elgin And Winter Garden Theatres', 43.653393796019586, -79.3785073962175, 'Theater'), ('Downtown Toronto', 43.6563221, -79.3809161, 'Banh Mi Boys', 43.659292, -79.381949, 'Sandwich Place'), ('Downtown Toronto', 43.6563221, -79.3809161, 'Pai', 43.64792310735613, -79.38857932631602, 'Thai Restaurant'), ('Downtown Toronto', 43.6563221, -79.3809161, 'Art Gallery of Ontario', 43.654002860337386, -79.39292172707437, 'Art Gallery'), ('Downtown Toronto', 43.6563221, -79.3809161, 'St. Lawrence Market (South Building)', 43.64874320223593, -79.37159697627527, 'Farmers Market')], [("Queen's Park", 43.659659, -79.3903399, 'Downtown Toronto', 43.65323167517444, -79.38529600606677, 'Neighborhood'), ("Queen's Park", 43.659659, -79.3903399, 'Art Gallery of Ontario', 43.654002860337386, -79.39292172707437, 'Art Gallery'), ("Queen's Park", 43.659659, -79.3903399, 'Hailed Coffee', 43.65883296982352, -79.38368351986598, 'Coffee Shop'), ("Queen's Park", 43.659659, -79.3903399, 'The Planet Traveler', 43.657202, -79.403568, 'Hostel'), ("Queen's Park", 43.659659, -79.3903399, 'Pai', 43.64792310735613, -79.38857932631602, 'Thai Restaurant'), ("Queen's Park", 43.659659, -79.3903399, 'Sonic Boom', 43.650858882525974, -79.39698455156606, 'Record Shop'), ("Queen's Park", 43.659659, -79.3903399, 'Byblos Toronto', 43.647615054171766, -79.3883810317138, 'Mediterranean Restaurant'), ("Queen's Park", 43.659659, -79.3903399, 'UNIQLO ユニクロ', 43.65591027779457, -79.38064099181345, 'Clothing Store'), ("Queen's Park", 43.659659, -79.3903399, 'Alo', 43.64857357946059, -79.39624325999638, 'French Restaurant'), ("Queen's Park", 43.659659, -79.3903399, 'Banh Mi Boys', 43.659292, -79.381949, 'Sandwich Place')], [('Etobicoke', 43.671459150000004, -79.55249206611668, 'Porta Via', 43.66344890472386, -79.58963815332794, 'Sandwich Place'), ('Etobicoke', 43.671459150000004, -79.55249206611668, 'Centennial Park', 43.6561537571146, -79.58753988924185, 'Park'), ('Etobicoke', 43.671459150000004, -79.55249206611668, 'Kingsway Fish & Chips', 43.647074, -79.513291, 'Seafood Restaurant'), ('Etobicoke', 43.671459150000004, -79.55249206611668, 'nodo', 43.665303321627505, -79.46562068004896, 'Italian Restaurant'), ('Etobicoke', 43.671459150000004, -79.55249206611668, 'Indigo', 43.61336504116866, -79.55798001820008, 'Bookstore'), ('Etobicoke', 43.671459150000004, -79.55249206611668, 'The Sweet Potato', 43.665405425838244, -79.4692294738573, 'Grocery Store'), ('Etobicoke', 43.671459150000004, -79.55249206611668, 'High Park', 43.6464791690415, -79.46342468261719, 'Park'), ('Etobicoke', 43.671459150000004, -79.55249206611668, 'SanRemo Bakery', 43.618542136521064, -79.49948481145465, 'Bakery'), ('Etobicoke', 43.671459150000004, -79.55249206611668, 'JOEY Sherway', 43.61309535370009, -79.55638480409485, 'Restaurant'), ('Etobicoke', 43.671459150000004, -79.55249206611668, 'Starbucks', 43.624654, -79.508217, 'Coffee Shop')], [('Scarborough', 43.773077, -79.257774, 'Charcoal Kebab House', 43.75734816869592, -79.23884167698144, 'Xinjiang Restaurant'), ('Scarborough', 43.773077, -79.257774, 'Toronto Pan Am Sports Centre', 43.79062336247366, -79.19386919338744, 'Athletics & Sports'), ('Scarborough', 43.773077, -79.257774, 'Scarborough Bluffs', 43.707779638489, -79.23723936080933, 'Beach'), ('Scarborough', 43.773077, -79.257774, 'Aga Khan Museum', 43.725105344569876, -79.33207588784165, 'History Museum'), ('Scarborough', 43.773077, -79.257774, 'Port Union Waterfront Park', 43.774967388607294, -79.13439273834229, 'Park'), ('Scarborough', 43.773077, -79.257774, 'Sunnybrook Park', 43.72231713758858, -79.35549634368492, 'Other Great Outdoors'), ('Scarborough', 43.773077, -79.257774, 'Edwards Gardens', 43.73357397323459, -79.35832879576286, 'Park'), ('Scarborough', 43.773077, -79.257774, "Lamanna's Bakery, Cafe & Fine Foods", 43.7979714045181, -79.14843228726039, 'Bakery'), ('Scarborough', 43.773077, -79.257774, 'Adamson Barbecue', 43.71245206628728, -79.35308334020402, 'BBQ Joint'), ('Scarborough', 43.773077, -79.257774, 'Maryam Hotel', 43.766961, -79.401199, 'Hotel')], [('East York', 43.699971000000005, -79.33251996261595, 'Adamson Barbecue', 43.71245206628728, -79.35308334020402, 'BBQ Joint'), ('East York', 43.699971000000005, -79.33251996261595, 'The Wren', 43.6824674097841, -79.32807925839526, 'American Restaurant'), ('East York', 43.699971000000005, -79.33251996261595, 'Rendez-Vous Restaurant Bar & Cafe', 43.68257026831632, -79.32754382523993, 'Ethiopian Restaurant'), ('East York', 43.699971000000005, -79.33251996261595, 'Serano Bakery', 43.683139418786745, -79.34653122970155, 'Bakery'), ('East York', 43.699971000000005, -79.33251996261595, 'Left Field Brewery', 43.67370495341353, -79.33031026982496, 'Brewery'), ('East York', 43.699971000000005, -79.33251996261595, 'CrossFit Toronto', 43.7080995557005, -79.35905992984772, 'Gym'), ('East York', 43.699971000000005, -79.33251996261595, 'Sunnybrook Park', 43.72231713758858, -79.35549634368492, 'Other Great Outdoors'), ('East York', 43.699971000000005, -79.33251996261595, 'Monarch Park Stadium', 43.6781436961028, -79.3240380491706, 'Soccer Field'), ('East York', 43.699971000000005, -79.33251996261595, 'Evergreen Brick Works', 43.68440081921645, -79.36524191949871, 'Historic Site'), ('East York', 43.699971000000005, -79.33251996261595, 'Aga Khan Museum', 43.725105344569876, -79.33207588784165, 'History Museum')], [('York', 43.67910515, -79.49118414007154, 'nodo', 43.665303321627505, -79.46562068004896, 'Italian Restaurant'), ('York', 43.67910515, -79.49118414007154, 'The Sweet Potato', 43.665405425838244, -79.4692294738573, 'Grocery Store'), ('York', 43.67910515, -79.49118414007154, 'Indie Alehouse', 43.66547472315272, -79.46528973047275, 'Gastropub'), ('York', 43.67910515, -79.49118414007154, 'Hole in the Wall', 43.66529624102518, -79.46511783107383, 'Bar'), ('York', 43.67910515, -79.49118414007154, 'High Park', 43.6464791690415, -79.46342468261719, 'Park'), ('York', 43.67910515, -79.49118414007154, 'Tre Mari Bakery', 43.676487, -79.44865, 'Café'), ('York', 43.67910515, -79.49118414007154, 'Cabin Fever', 43.65544217064788, -79.45663939393057, 'Bar'), ('York', 43.67910515, -79.49118414007154, 'Hale Coffee', 43.665531698018086, -79.44994926452637, 'Coffee Shop'), ('York', 43.67910515, -79.49118414007154, 'The Greater Good Bar', 43.669409, -79.439267, 'Bar'), ('York', 43.67910515, -79.49118414007154, 'Pizzeria Defina', 43.648722638277675, -79.44985455915354, 'Pizza Place')], [('East Toronto', 43.6247901, -79.3934918, 'Toronto Islands', 43.62211231383369, -79.37849521636963, 'Park'), ('East Toronto', 43.6247901, -79.3934918, 'Steam Whistle Brewing', 43.641751628677554, -79.38708862583252, 'Brewery'), ('East Toronto', 43.6247901, -79.3934918, 'Roundhouse Park', 43.64174513889102, -79.38427882922346, 'Park'), ('East Toronto', 43.6247901, -79.3934918, 'Harbourfront Centre', 43.63855578926675, -79.38319016270458, 'Performing Arts Venue'), ('East Toronto', 43.6247901, -79.3934918, 'SOMA chocolatemaker', 43.64532829235515, -79.39571438648944, 'Dessert Shop'), ('East Toronto', 43.6247901, -79.3934918, 'Delta Hotels by Marriott Toronto', 43.6428819, -79.3839491, 'Hotel'), ('East Toronto', 43.6247901, -79.3934918, 'Harbour Square Park', 43.63925269130776, -79.37839547902978, 'Park'), ('East Toronto', 43.6247901, -79.3934918, 'Porchetta & Co', 43.64466404008139, -79.39881259826922, 'Sandwich Place'), ('East Toronto', 43.6247901, -79.3934918, 'Lake Ontario', 43.63894493157648, -79.37966465950012, 'Lake'), ('East Toronto', 43.6247901, -79.3934918, 'Toronto Music Garden', 43.636901770652464, -79.39393331122407, 'Garden')], [('West Toronto', 43.653963, -79.387207, 'Downtown Toronto', 43.65323167517444, -79.38529600606677, 'Neighborhood'), ('West Toronto', 43.653963, -79.387207, 'Art Gallery of Ontario', 43.654002860337386, -79.39292172707437, 'Art Gallery'), ('West Toronto', 43.653963, -79.387207, 'Pai', 43.64792310735613, -79.38857932631602, 'Thai Restaurant'), ('West Toronto', 43.653963, -79.387207, 'Byblos Toronto', 43.647615054171766, -79.3883810317138, 'Mediterranean Restaurant'), ('West Toronto', 43.653963, -79.387207, 'SOMA chocolatemaker', 43.64532829235515, -79.39571438648944, 'Dessert Shop'), ('West Toronto', 43.653963, -79.387207, 'Richmond Station', 43.65156872562108, -79.37926594000328, 'American Restaurant'), ('West Toronto', 43.653963, -79.387207, 'Alo', 43.64857357946059, -79.39624325999638, 'French Restaurant'), ('West Toronto', 43.653963, -79.387207, 'Sonic Boom', 43.650858882525974, -79.39698455156606, 'Record Shop'), ('West Toronto', 43.653963, -79.387207, 'UNIQLO ユニクロ', 43.65591027779457, -79.38064099181345, 'Clothing Store'), ('West Toronto', 43.653963, -79.387207, 'Hailed Coffee', 43.65883296982352, -79.38368351986598, 'Coffee Shop')], [('Central Toronto', 43.653963, -79.387207, 'Downtown Toronto', 43.65323167517444, -79.38529600606677, 'Neighborhood'), ('Central Toronto', 43.653963, -79.387207, 'Art Gallery of Ontario', 43.654002860337386, -79.39292172707437, 'Art Gallery'), ('Central Toronto', 43.653963, -79.387207, 'Pai', 43.64792310735613, -79.38857932631602, 'Thai Restaurant'), ('Central Toronto', 43.653963, -79.387207, 'Byblos Toronto', 43.647615054171766, -79.3883810317138, 'Mediterranean Restaurant'), ('Central Toronto', 43.653963, -79.387207, 'SOMA chocolatemaker', 43.64532829235515, -79.39571438648944, 'Dessert Shop'), ('Central Toronto', 43.653963, -79.387207, 'Richmond Station', 43.65156872562108, -79.37926594000328, 'American Restaurant'), ('Central Toronto', 43.653963, -79.387207, 'Alo', 43.64857357946059, -79.39624325999638, 'French Restaurant'), ('Central Toronto', 43.653963, -79.387207, 'Sonic Boom', 43.650858882525974, -79.39698455156606, 'Record Shop'), ('Central Toronto', 43.653963, -79.387207, 'UNIQLO ユニクロ', 43.65591027779457, -79.38064099181345, 'Clothing Store'), ('Central Toronto', 43.653963, -79.387207, 'Hailed Coffee', 43.65883296982352, -79.38368351986598, 'Coffee Shop')], [('Mississauga', 43.6668555, -79.5879563, 'Porta Via', 43.66344890472386, -79.58963815332794, 'Sandwich Place'), ('Mississauga', 43.6668555, -79.5879563, 'Centennial Park', 43.6561537571146, -79.58753988924185, 'Park'), ('Mississauga', 43.6668555, -79.5879563, 'Indigo', 43.61336504116866, -79.55798001820008, 'Bookstore'), ('Mississauga', 43.6668555, -79.5879563, 'Kingsway Fish & Chips', 43.647074, -79.513291, 'Seafood Restaurant'), ('Mississauga', 43.6668555, -79.5879563, 'JOEY Sherway', 43.61309535370009, -79.55638480409485, 'Restaurant'), ('Mississauga', 43.6668555, -79.5879563, 'SanRemo Bakery', 43.618542136521064, -79.49948481145465, 'Bakery'), ('Mississauga', 43.6668555, -79.5879563, 'Starbucks', 43.624654, -79.508217, 'Coffee Shop'), ('Mississauga', 43.6668555, -79.5879563, 'High Park', 43.6464791690415, -79.46342468261719, 'Park'), ('Mississauga', 43.6668555, -79.5879563, 'Humber Bay Park', 43.6223956417084, -79.47838891858714, 'Park'), ('Mississauga', 43.6668555, -79.5879563, 'nodo', 43.665303321627505, -79.46562068004896, 'Italian Restaurant')]]
         Neighborhood  Neighborhood Latitude  Neighborhood Longitude  \
0          North York              43.754326              -79.449117   
1          North York              43.754326              -79.449117   
2          North York              43.754326              -79.449117   
3          North York              43.754326              -79.449117   
4          North York              43.754326              -79.449117   
5          North York              43.754326              -79.449117   
6          North York              43.754326              -79.449117   
7          North York              43.754326              -79.449117   
8          North York              43.754326              -79.449117   
9          North York              43.754326              -79.449117   
10   Downtown Toronto              43.656322              -79.380916   
11   Downtown Toronto              43.656322              -79.380916   
12   Downtown Toronto              43.656322              -79.380916   
13   Downtown Toronto              43.656322              -79.380916   
14   Downtown Toronto              43.656322              -79.380916   
15   Downtown Toronto              43.656322              -79.380916   
16   Downtown Toronto              43.656322              -79.380916   
17   Downtown Toronto              43.656322              -79.380916   
18   Downtown Toronto              43.656322              -79.380916   
19   Downtown Toronto              43.656322              -79.380916   
20       Queen's Park              43.659659              -79.390340   
21       Queen's Park              43.659659              -79.390340   
22       Queen's Park              43.659659              -79.390340   
23       Queen's Park              43.659659              -79.390340   
24       Queen's Park              43.659659              -79.390340   
25       Queen's Park              43.659659              -79.390340   
26       Queen's Park              43.659659              -79.390340   
27       Queen's Park              43.659659              -79.390340   
28       Queen's Park              43.659659              -79.390340   
29       Queen's Park              43.659659              -79.390340   
30          Etobicoke              43.671459              -79.552492   
31          Etobicoke              43.671459              -79.552492   
32          Etobicoke              43.671459              -79.552492   
33          Etobicoke              43.671459              -79.552492   
34          Etobicoke              43.671459              -79.552492   
35          Etobicoke              43.671459              -79.552492   
36          Etobicoke              43.671459              -79.552492   
37          Etobicoke              43.671459              -79.552492   
38          Etobicoke              43.671459              -79.552492   
39          Etobicoke              43.671459              -79.552492   
40        Scarborough              43.773077              -79.257774   
41        Scarborough              43.773077              -79.257774   
42        Scarborough              43.773077              -79.257774   
43        Scarborough              43.773077              -79.257774   
44        Scarborough              43.773077              -79.257774   
45        Scarborough              43.773077              -79.257774   
46        Scarborough              43.773077              -79.257774   
47        Scarborough              43.773077              -79.257774   
48        Scarborough              43.773077              -79.257774   
49        Scarborough              43.773077              -79.257774   
50          East York              43.699971              -79.332520   
51          East York              43.699971              -79.332520   
52          East York              43.699971              -79.332520   
53          East York              43.699971              -79.332520   
54          East York              43.699971              -79.332520   
55          East York              43.699971              -79.332520   
56          East York              43.699971              -79.332520   
57          East York              43.699971              -79.332520   
58          East York              43.699971              -79.332520   
59          East York              43.699971              -79.332520   
60               York              43.679105              -79.491184   
61               York              43.679105              -79.491184   
62               York              43.679105              -79.491184   
63               York              43.679105              -79.491184   
64               York              43.679105              -79.491184   
65               York              43.679105              -79.491184   
66               York              43.679105              -79.491184   
67               York              43.679105              -79.491184   
68               York              43.679105              -79.491184   
69               York              43.679105              -79.491184   
70       East Toronto              43.624790              -79.393492   
71       East Toronto              43.624790              -79.393492   
72       East Toronto              43.624790              -79.393492   
73       East Toronto              43.624790              -79.393492   
74       East Toronto              43.624790              -79.393492   
75       East Toronto              43.624790              -79.393492   
76       East Toronto              43.624790              -79.393492   
77       East Toronto              43.624790              -79.393492   
78       East Toronto              43.624790              -79.393492   
79       East Toronto              43.624790              -79.393492   
80       West Toronto              43.653963              -79.387207   
81       West Toronto              43.653963              -79.387207   
82       West Toronto              43.653963              -79.387207   
83       West Toronto              43.653963              -79.387207   
84       West Toronto              43.653963              -79.387207   
85       West Toronto              43.653963              -79.387207   
86       West Toronto              43.653963              -79.387207   
87       West Toronto              43.653963              -79.387207   
88       West Toronto              43.653963              -79.387207   
89       West Toronto              43.653963              -79.387207   
90    Central Toronto              43.653963              -79.387207   
91    Central Toronto              43.653963              -79.387207   
92    Central Toronto              43.653963              -79.387207   
93    Central Toronto              43.653963              -79.387207   
94    Central Toronto              43.653963              -79.387207   
95    Central Toronto              43.653963              -79.387207   
96    Central Toronto              43.653963              -79.387207   
97    Central Toronto              43.653963              -79.387207   
98    Central Toronto              43.653963              -79.387207   
99    Central Toronto              43.653963              -79.387207   
100       Mississauga              43.666855              -79.587956   
101       Mississauga              43.666855              -79.587956   
102       Mississauga              43.666855              -79.587956   
103       Mississauga              43.666855              -79.587956   
104       Mississauga              43.666855              -79.587956   
105       Mississauga              43.666855              -79.587956   
106       Mississauga              43.666855              -79.587956   
107       Mississauga              43.666855              -79.587956   
108       Mississauga              43.666855              -79.587956   
109       Mississauga              43.666855              -79.587956   

                                    Venue  Venue Latitude  Venue Longitude  \
0                         Earl Bales Park       43.753043       -79.436228   
1                               Nordstrom       43.726054       -79.449443   
2                            Maryam Hotel       43.766961       -79.401199   
3                           Kinka Izakaya       43.760161       -79.409827   
4                          Crate & Barrel       43.726584       -79.452661   
5                      Auberge du Pommier       43.746962       -79.407879   
6                      Amadeus Fine Cakes       43.802582       -79.448743   
7                               The Abbot       43.703688       -79.413485   
8                          Cedarvale Park       43.692535       -79.428705   
9                           Sherwood Park       43.716551       -79.387776   
10                       Downtown Toronto       43.653232       -79.385296   
11                            UNIQLO ユニクロ       43.655910       -79.380641   
12                          Hailed Coffee       43.658833       -79.383684   
13                       Richmond Station       43.651569       -79.379266   
14                      Fahrenheit Coffee       43.652384       -79.372719   
15       Elgin And Winter Garden Theatres       43.653394       -79.378507   
16                           Banh Mi Boys       43.659292       -79.381949   
17                                    Pai       43.647923       -79.388579   
18                 Art Gallery of Ontario       43.654003       -79.392922   
19   St. Lawrence Market (South Building)       43.648743       -79.371597   
20                       Downtown Toronto       43.653232       -79.385296   
21                 Art Gallery of Ontario       43.654003       -79.392922   
22                          Hailed Coffee       43.658833       -79.383684   
23                    The Planet Traveler       43.657202       -79.403568   
24                                    Pai       43.647923       -79.388579   
25                             Sonic Boom       43.650859       -79.396985   
26                         Byblos Toronto       43.647615       -79.388381   
27                            UNIQLO ユニクロ       43.655910       -79.380641   
28                                    Alo       43.648574       -79.396243   
29                           Banh Mi Boys       43.659292       -79.381949   
30                              Porta Via       43.663449       -79.589638   
31                        Centennial Park       43.656154       -79.587540   
32                  Kingsway Fish & Chips       43.647074       -79.513291   
33                                   nodo       43.665303       -79.465621   
34                                 Indigo       43.613365       -79.557980   
35                       The Sweet Potato       43.665405       -79.469229   
36                              High Park       43.646479       -79.463425   
37                         SanRemo Bakery       43.618542       -79.499485   
38                           JOEY Sherway       43.613095       -79.556385   
39                              Starbucks       43.624654       -79.508217   
40                   Charcoal Kebab House       43.757348       -79.238842   
41           Toronto Pan Am Sports Centre       43.790623       -79.193869   
42                     Scarborough Bluffs       43.707780       -79.237239   
43                        Aga Khan Museum       43.725105       -79.332076   
44             Port Union Waterfront Park       43.774967       -79.134393   
45                        Sunnybrook Park       43.722317       -79.355496   
46                        Edwards Gardens       43.733574       -79.358329   
47    Lamanna's Bakery, Cafe & Fine Foods       43.797971       -79.148432   
48                       Adamson Barbecue       43.712452       -79.353083   
49                           Maryam Hotel       43.766961       -79.401199   
50                       Adamson Barbecue       43.712452       -79.353083   
51                               The Wren       43.682467       -79.328079   
52      Rendez-Vous Restaurant Bar & Cafe       43.682570       -79.327544   
53                          Serano Bakery       43.683139       -79.346531   
54                     Left Field Brewery       43.673705       -79.330310   
55                       CrossFit Toronto       43.708100       -79.359060   
56                        Sunnybrook Park       43.722317       -79.355496   
57                   Monarch Park Stadium       43.678144       -79.324038   
58                  Evergreen Brick Works       43.684401       -79.365242   
59                        Aga Khan Museum       43.725105       -79.332076   
60                                   nodo       43.665303       -79.465621   
61                       The Sweet Potato       43.665405       -79.469229   
62                         Indie Alehouse       43.665475       -79.465290   
63                       Hole in the Wall       43.665296       -79.465118   
64                              High Park       43.646479       -79.463425   
65                        Tre Mari Bakery       43.676487       -79.448650   
66                            Cabin Fever       43.655442       -79.456639   
67                            Hale Coffee       43.665532       -79.449949   
68                   The Greater Good Bar       43.669409       -79.439267   
69                        Pizzeria Defina       43.648723       -79.449855   
70                        Toronto Islands       43.622112       -79.378495   
71                  Steam Whistle Brewing       43.641752       -79.387089   
72                        Roundhouse Park       43.641745       -79.384279   
73                    Harbourfront Centre       43.638556       -79.383190   
74                    SOMA chocolatemaker       43.645328       -79.395714   
75       Delta Hotels by Marriott Toronto       43.642882       -79.383949   
76                    Harbour Square Park       43.639253       -79.378395   
77                         Porchetta & Co       43.644664       -79.398813   
78                           Lake Ontario       43.638945       -79.379665   
79                   Toronto Music Garden       43.636902       -79.393933   
80                       Downtown Toronto       43.653232       -79.385296   
81                 Art Gallery of Ontario       43.654003       -79.392922   
82                                    Pai       43.647923       -79.388579   
83                         Byblos Toronto       43.647615       -79.388381   
84                    SOMA chocolatemaker       43.645328       -79.395714   
85                       Richmond Station       43.651569       -79.379266   
86                                    Alo       43.648574       -79.396243   
87                             Sonic Boom       43.650859       -79.396985   
88                            UNIQLO ユニクロ       43.655910       -79.380641   
89                          Hailed Coffee       43.658833       -79.383684   
90                       Downtown Toronto       43.653232       -79.385296   
91                 Art Gallery of Ontario       43.654003       -79.392922   
92                                    Pai       43.647923       -79.388579   
93                         Byblos Toronto       43.647615       -79.388381   
94                    SOMA chocolatemaker       43.645328       -79.395714   
95                       Richmond Station       43.651569       -79.379266   
96                                    Alo       43.648574       -79.396243   
97                             Sonic Boom       43.650859       -79.396985   
98                            UNIQLO ユニクロ       43.655910       -79.380641   
99                          Hailed Coffee       43.658833       -79.383684   
100                             Porta Via       43.663449       -79.589638   
101                       Centennial Park       43.656154       -79.587540   
102                                Indigo       43.613365       -79.557980   
103                 Kingsway Fish & Chips       43.647074       -79.513291   
104                          JOEY Sherway       43.613095       -79.556385   
105                        SanRemo Bakery       43.618542       -79.499485   
106                             Starbucks       43.624654       -79.508217   
107                             High Park       43.646479       -79.463425   
108                       Humber Bay Park       43.622396       -79.478389   
109                                  nodo       43.665303       -79.465621   

               Venue Category  
0                        Park  
1              Clothing Store  
2                       Hotel  
3         Japanese Restaurant  
4      Furniture / Home Store  
5           French Restaurant  
6                Dessert Shop  
7                   Gastropub  
8                       Field  
9                        Park  
10               Neighborhood  
11             Clothing Store  
12                Coffee Shop  
13        American Restaurant  
14                Coffee Shop  
15                    Theater  
16             Sandwich Place  
17            Thai Restaurant  
18                Art Gallery  
19             Farmers Market  
20               Neighborhood  
21                Art Gallery  
22                Coffee Shop  
23                     Hostel  
24            Thai Restaurant  
25                Record Shop  
26   Mediterranean Restaurant  
27             Clothing Store  
28          French Restaurant  
29             Sandwich Place  
30             Sandwich Place  
31                       Park  
32         Seafood Restaurant  
33         Italian Restaurant  
34                  Bookstore  
35              Grocery Store  
36                       Park  
37                     Bakery  
38                 Restaurant  
39                Coffee Shop  
40        Xinjiang Restaurant  
41         Athletics & Sports  
42                      Beach  
43             History Museum  
44                       Park  
45       Other Great Outdoors  
46                       Park  
47                     Bakery  
48                  BBQ Joint  
49                      Hotel  
50                  BBQ Joint  
51        American Restaurant  
52       Ethiopian Restaurant  
53                     Bakery  
54                    Brewery  
55                        Gym  
56       Other Great Outdoors  
57               Soccer Field  
58              Historic Site  
59             History Museum  
60         Italian Restaurant  
61              Grocery Store  
62                  Gastropub  
63                        Bar  
64                       Park  
65                       Café  
66                        Bar  
67                Coffee Shop  
68                        Bar  
69                Pizza Place  
70                       Park  
71                    Brewery  
72                       Park  
73      Performing Arts Venue  
74               Dessert Shop  
75                      Hotel  
76                       Park  
77             Sandwich Place  
78                       Lake  
79                     Garden  
80               Neighborhood  
81                Art Gallery  
82            Thai Restaurant  
83   Mediterranean Restaurant  
84               Dessert Shop  
85        American Restaurant  
86          French Restaurant  
87                Record Shop  
88             Clothing Store  
89                Coffee Shop  
90               Neighborhood  
91                Art Gallery  
92            Thai Restaurant  
93   Mediterranean Restaurant  
94               Dessert Shop  
95        American Restaurant  
96          French Restaurant  
97                Record Shop  
98             Clothing Store  
99                Coffee Shop  
100            Sandwich Place  
101                      Park  
102                 Bookstore  
103        Seafood Restaurant  
104                Restaurant  
105                    Bakery  
106               Coffee Shop  
107                      Park  
108                      Park  
109

In [33]:
list_centers=[]

for i in range(len(toronto_venues["Venue Longitude"])):
    list_h=[]
    list_h.append(toronto_venues["Venue Latitude"][i])
    list_h.append(toronto_venues["Venue Longitude"][i])
    list_centers.append(list_h)
list_centers[:10]

NameError: name 'toronto_venues' is not defined